### PR TITLE
Ensure thread archive state is also synced when syncing channels

### LINKED ISSUE #153: Threads archive state is not synchronized after start
Currently the thread archive state is only synchronized on startup. This means
that the `archived` flag on threads in the database slowly runs out of sync.

We should probably listen for
[`on_thread_update`](https://discordpy.readthedocs.io/en/stable/api.html#discord.on_thread_update)
or
[`on_raw_thread_update`](https://discordpy.readthedocs.io/en/stable/api.html#discord.on_raw_thread_update)
here and update the database accordingly.

### PR DESCRIPTION
**Pull request by ChrisLovering**
Closes #153

### CODE REVIEW

File: metricity/exts/event_listeners/_syncer_utils.py [hunk 1]
```diff
@@ -117,7 +117,8 @@ async def sync_channels(bot: Bot, guild: discord.Guild) -> None:
                 if thread.parent.category.id in BotConfig.ignore_categories:
                     continue
             else:
-                # This is a forum channel, not currently supported by Discord.py. Ignore it.
+                # This use to happen for forum channels, when not supported by Discord.py.
+                # This case should no longer be possible, but will be kept for future possibilities.
```
[the remark this answers is not in the document]
Comment (ChrisLovering) [reply metricity/exts/event_listeners/_syncer_utils.py:121]: That's true, I'll remove the case entirely

[the remark this answers is not in the document]
Comment (ChrisLovering) [reply metricity/exts/event_listeners/_syncer_utils.py]: https://github.com/python-discord/metricity/compare/b49dd53e7ecfa184c4e45c799e29709617b72df8..9219d8435014fdfedc7ad06888f976d61314ab4f

**Comment by ChrisLovering** [comment]
> Thanks! I assume the bug was the lacking support for forum channels in older versions then?

Nah, the bug was that `sync_thread_archive_state` wasn't called when syncing channels, it was only called on startup.

The other part of this change isn't relevant to this issue, but added confusion to the function due to now being relevant anymore now that d.py supports forum channels

**Comment by wookie184** [comment]
(and `sync_thread_archive_state` is needed because the `db_thread.archived = thread.archived` line was not sufficient as archived threads are not in `guild.threads`)

**Review by wookie184 — APPROVED** [review]
lgtm

**Pull request merged**